sources/packages/backend/apps/api/src/route-controllers/assessment/assessment.institutions.controller.ts

```diff
@@ -0,0 +1,71 @@
+import { Controller, Get, Param, ParseIntPipe } from "@nestjs/common";
+import BaseController from "../BaseController";
+import {
+  AllowAuthorizedParty,
+  HasStudentDataAccess,
+  IsBCPublicInstitution,
+} from "../../auth/decorators";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { ClientTypeBaseRoute } from "../../types";
+import {
+  AssessmentHistorySummaryAPIOutDTO,
+  RequestAssessmentSummaryAPIOutDTO,
+} from "./models/assessment.dto";
+import { ApiTags } from "@nestjs/swagger";
+import { AssessmentControllerService } from "./assessment.controller.service";
+
+/**
+ * Assessment controller for institutions.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("assessment")
+@IsBCPublicInstitution()
+@ApiTags(`${ClientTypeBaseRoute.Institution}-assessment`)
+export class AssessmentInstitutionsController extends BaseController {
+  constructor(
+    private readonly assessmentControllerService: AssessmentControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get all pending and declined requests related to an application which would result
+   * a new assessment when that request is approved.
+   * @param studentId student id.
+   * @param applicationId application id.
+   * @returns assessment requests or exceptions for the student application.
+   */
+  @HasStudentDataAccess("studentId")
```

Comment:
I saw the discussion in other PRs about having it at the class level or at the method level.
I would say that we can keep it at the class level and move to the method level if we ever introduce a method that does not need it.
It would be less error-prone for a developer in the future to add a new endpoint.

Reply:
I like this way, to see the decorator with parameters near the controller, and for the same reason that I replied in the other PR. But if the team prefers the other way, I can change

Reply:
I see the point, Ann, to be clear why I like to have it at the class level is because a developer adding a new endpoint can forget to add the decorator and the endpoint can be exposed. I would be ok in one way or another, but in terms of security, I would prefer it at the class level.

Reply:
That was exactly my point.